Makefile

```diff
@@ -1,163 +1,13 @@
-
-jq/install: JQ_VERSION ?= latest
-jq/install: | $(BIN_DIR)
-	@ $(MAKE) install/gh-release/$(@D) FILENAME="$(BIN_DIR)/$(@D)" OWNER=stedolan REPO=$(@D) VERSION=$(JQ_VERSION) QUERY='.name | endswith("$(OS)64")'
-
-shellcheck/install: SHELLCHECK_VERSION ?= latest
-shellcheck/install: SHELLCHECK_URL ?= https://storage.googleapis.com/shellcheck/shellcheck-${SHELLCHECK_VERSION}.linux.x86_64.tar.xz
-shellcheck/install: $(BIN_DIR) guard/program/xz
-	$(CURL) $(SHELLCHECK_URL) | tar -xJv
-	mv $(@D)-*/$(@D) $(BIN_DIR)
-	rm -rf $(@D)-*
-	$(@D) --version
-
-terraform/lint: | guard/program/terraform
-	@ echo "[$@]: Linting Terraform files..."
-	terraform fmt -check=true -diff=true
-	@ echo "[$@]: Terraform files PASSED lint test!"
-
-sh/%: FIND_SH := find . $(FIND_EXCLUDES) -name '*.sh' -type f -print0
-sh/lint: | guard/program/shellcheck
-	@ echo "[$@]: Linting shell scripts..."
-	$(FIND_SH) | $(XARGS) shellcheck {}
-	@ echo "[$@]: Shell scripts PASSED lint test!"
-
-json/%: FIND_JSON := find . $(FIND_EXCLUDES) -name '*.json' -type f
-json/lint: | guard/program/jq
-	@ echo "[$@]: Linting JSON files..."
-	$(FIND_JSON) | $(XARGS) bash -c 'cmp {} <(jq --indent 4 -S . {}) || (echo "[{}]: Failed JSON Lint Test"; exit 1)'
-	@ echo "[$@]: JSON files PASSED lint test!"
-
-json/format: | guard/program/jq
-	@ echo "[$@]: Formatting JSON files..."
-	$(FIND_JSON) | $(XARGS) bash -c 'echo "$$(jq --indent 4 -S . "{}")" > "{}"'
-	@ echo "[$@]: Successfully formatted JSON files!"
-
-terraform/format: | guard/program/terraform
-	@ echo "[$@]: Formatting Terraform files..."
-	terraform fmt -recursive
-	@ echo "[$@]: Successfully formatted terraform files!"
-
-tfdocs-awk/install: $(BIN_DIR)
-tfdocs-awk/install: ARCHIVE := https://github.com/plus3it/tfdocs-awk/archive/0.0.0.tar.gz
-tfdocs-awk/install:
-	$(CURL) $(ARCHIVE) | tar -C $(BIN_DIR) --strip-components=1 --wildcards '*.sh' --wildcards '*.awk' -xzvf -
-
-docs/generate: | tfdocs-awk/install guard/program/terraform-docs
-	@ echo "[$@]: Creating documentation files.."
-	@ bash -eu -o pipefail autodocs.sh -g
-	@ echo "[$@]: Documentation generated!"
-
-docs/lint: | tfdocs-awk/install guard/program/terraform-docs
-	@ echo "[$@] Linting documentation files.."
-	@ bash -eu -o pipefail autodocs.sh -l
-	@ echo "[$@] documentation linting complete!"
-
-python/lint: | guard/program/black
-	@ echo "[$@]: Linting Python files..."
-	black --check . --exclude vendor/
-	@ echo "[$@]: Python files PASSED lint test!"
-
-python/format: | guard/program/black
-	@ echo "[$@]: Formatting Python files..."
-	black . --exclude vendor/
-	@ echo "[$@]: Successfully formatted Python files!"
-
-terratest/install: | guard/program/go
-	cd tests && go mod init terraform-aws-tardigrade-config-rules/tests
-	cd tests && go build ./...
-	cd tests && go mod tidy
-
-terratest/test: | guard/program/go
-	cd tests && go test -timeout 20m
-
-test: terratest/test
-
-clean:
+clean::
```

Comment:
What's with the double `:`

Reply:
another gnu make quirk. error generated without the double `::`
`Makefile:5: *** target file 'clean' has both : and :: entries.  Stop.`

Reply:
https://www.gnu.org/software/make/manual/html_node/Double_002dColon.html

Reply:
from my understanding after reading that doc using `<target>::` is treading into "unintended actions" territory. updated to use `<target>/%:` format

Reply:
It just means you have more than one `clean` target, and they should both execute independently when calling `make clean`.